### PR TITLE
Status

### PR DESCRIPTION
**Pull request by BenjixD**
- Adding Concrete Status Ailments

Please See:
- DefenseStatusAilment
- ParalyzeStatusAilment
- PoisonStatusAilment

For Paralyze, the status actually replaces the queued action at defined % chance. This action can be found in `Actions/Paralyzed.cs`.

For Actions which test these ailments, see:
- Defend.cs
- ThunderOne.cs
- PoisonAll.cs

### CODE REVIEW

File: Assets/_Scripts/Actions/PoisonAllTest.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if (!BasicValidation(splitCommand)) {
+            return false;
+        }
+        List<int> targetIds = this.GetAllPossibleTargets(user).Map((FightingEntity target) => target.targetId );
+        user.SetQueuedAction(new QueuedAction(user, this, targetIds));
+        return true;
+    }
+
+    public override FightResult ApplyEffect(FightingEntity user, List<FightingEntity> targets) {
+        PlayerStats before, after;
```
Comment (antu3199) [inline Assets/_Scripts/Actions/PoisonAllTest.cs:21]: Doesn't have to be done in this pr, but can you think of a way to add helper functions so we don't have to rewrite this code for every action?

File: Assets/_Scripts/Actions/ThunderOneTest.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if (targetId == -1) {
+            return false;
+        }
+
+        user.SetQueuedAction(new QueuedAction(user, this, new List<int>{ targetId }));
+        return true;
+    }
+
+    public override FightResult ApplyEffect(FightingEntity user, List<FightingEntity> targets) {
+        PlayerStats before, after;
```
Comment (antu3199) [inline Assets/_Scripts/Actions/ThunderOneTest.cs:26]: See above note

File: Assets/_Scripts/Character/StatusAilment/StatusAilment.cs [hunk 3]
```diff
@@ -2,10 +2,11 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public abstract class StatusAilment {
+public abstract class StatusAilment : ScriptableObject {
 	public string name;
 	public int duration;
 	public int level;
+	[Header("Phase to trigger Status Ailment")]
```
Comment (antu3199) [inline Assets/_Scripts/Character/StatusAilment/StatusAilment.cs:9]: Minor, but I don't think it's common to add a header for a variable description. Headers are to group variables into categories. I think Tooltips are what you want to use.

File: Assets/_Scripts/BattleFight.cs [hunk 4]
```diff
@@ -53,9 +62,9 @@ public class BattleFight
     }
 
     public IEnumerator DoFight() {
-        if (this.fighter.isEnemy()) {
-            this.getEnemyAction();
-        }
+        // if (this.fighter.isEnemy()) {
```
Comment (antu3199) [inline Assets/_Scripts/BattleFight.cs:65]: Where has this been moved to?

Comment (BenjixD) [reply Assets/_Scripts/BattleFight.cs:65]: no where. It was being called twice.

Comment (antu3199) [reply Assets/_Scripts/BattleFight.cs:65]: :OOO

File: Assets/_Scripts/Character/StatusAilment/StatusAilment.cs (hunk 3 above)
Comment (BenjixD) [reply Assets/_Scripts/Character/StatusAilment/StatusAilment.cs:9]: good point

**Pull request merged**